public/app/plugins/datasource/cloudwatch/query-runner/CloudWatchAnnotationQueryRunner.test.ts

```diff
@@ -0,0 +1,41 @@
+import { setupMockedAnnotationQueryRunner } from '../__mocks__/AnnotationQueryRunner';
+import { namespaceVariable, regionVariable } from '../__mocks__/CloudWatchDataSource';
+import { CloudWatchAnnotationQuery } from '../types';
+
+describe('CloudWatchAnnotationQueryRunner', () => {
+  const queries: CloudWatchAnnotationQuery[] = [
+    {
+      actionPrefix: '',
+      alarmNamePrefix: '',
+      datasource: { type: 'cloudwatch' },
+      dimensions: { InstanceId: 'i-12345678' },
+      matchExact: true,
+      metricName: 'CPUUtilization',
+      period: '300',
+      prefixMatching: false,
+      queryMode: 'Annotations',
+      refId: 'Anno',
+      namespace: `$${namespaceVariable.name}`,
+      region: `$${regionVariable.name}`,
+      statistic: 'Average',
+    },
+  ];
+
+  it('should issue the correct query', async () => {
+    const { runner, fetchMock, request } = setupMockedAnnotationQueryRunner({
+      variables: [namespaceVariable, regionVariable],
+    });
+    await expect(runner.handleAnnotationQuery(queries, request)).toEmitValuesWith(() => {
+      expect(fetchMock.mock.calls[0][0].data.queries[0]).toMatchObject(
+        expect.objectContaining({
+          region: regionVariable.current.value,
```

Comment:
these nested expects are interesting, does this work are they each necessary? I sort of would expect only the top level expect would be necessary but I might be missing stuff.

Reply:
Probably not! Feels like these tests can be improved a lot, but I've been trying to not touch them unless they are for example duplicating test fixtures/test setup code and such. Is it ok if we tackle refactoring existing tests in another PR?